docs/Gopkg.toml.md

```diff
@@ -0,0 +1,103 @@
+# Gopkg.toml
+
+## `required`
+`required` lists a set of packages (not projects) that must be included in Gopkg.lock. This list is merged with the set of packages imported by the current project. Use it when your project needs a package it doesn't explicitly import - including "main" packages.
+ ```toml
+required = ["github.com/user/thing/cmd/thing"]
+```
+
+## `ignored`
+`ignored` lists a set of packages (not projects) that are ignored when dep statically analyzes source code. Ignored packages can be in this project, or in a dependency.
+```toml
+ignored = ["github.com/user/project/badpkg"]
+```
+
+## `metadata`
+`metadata` can exist at the root as well as under `constraint` and `override` declarations. 
+
+`metadata` declarations are ignored by dep and are meant for usage by other independent systems.
+
+A `metadata` declaration at the root defines metadata about the project itself. While a `metadata` declaration under a `constraint` or an `override` defines metadata about that `constraint` or `override`.
+```toml
+[metadata]
+key1 = "value that convey data to other systems"
+system1-data = "value that is used by a system"
+system2-data = "value that is used by another system"
+```
+
+## `constraint`
+A `constraint` provides rules for how directly imported projects may be incorporated into the dependency graph. They are respected by dep whether coming from the Gopkg.toml of the current project or a dependency.
```

Comment:
My preference is to use consistent terminology, such as "direct dependency", and the first time it is used, link to our [FAQ](https://github.com/golang/dep/blob/master/FAQ.md#what-is-a-direct-or-transitive-dependency) for any user for whom it is unfamiliar.